### ISSUE TITLE
Use C++20 library feature tests

### ISSUE THREAD

**Issue by purpleKarrot** [nonstd-lite/expected-lite#48]
The following code breaks when it is compiled with a C++20 capable compiler without using a C++20 capable standard library:
```cpp
#if nsel_CPP20_OR_GREATER

using std::remove_cvref;

#else
```

Please don't assume that C++20 library features are available depending on the language standard that was selected with `-std=`.
Instead, check for each feature individually, as in

```cpp
#if defined(__cpp_lib_remove_cvref)

using std::remove_cvref;

#else
```

See the list of feature tests here: https://en.cppreference.com/w/cpp/feature_test

**Comment by striezel** [comment]
Looks like this issue has been fixed in commit 34524d46e538f1e6ed114d8f8409f7cd173d96e6 / [release 0.6.2](https://github.com/martinmoene/expected-lite/releases/tag/v0.6.2), so this issue could be closed.

**Comment by martinmoene** [comment]
Thanks @striezel !
